File: TEK3/Java/Java_area_2017/src/main/java/com/area/modules/action/twitter/CheckTweet.java
package com.area.modules.action.twitter;

import com.area.modules.Action;
import com.area.modules.Reaction;

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.net.URL;
import java.net.URLConnection;
import java.util.*;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class CheckTweet implements Action {
    Map<String, String> fields = new HashMap<String, String>();
    Reaction r = null;

    URLConnection connection = null;

    int nbTweet = 0;

    public List<String> getFields() {
        return (new ArrayList<String>() {{
            add("username");
        }});
    }
    public void setField(Map<String, String> fields) {
        this.fields = fields;
        this.nbTweet = getNbTweet();
    }
    public void addReaction(Reaction r) {
        this.r = r;
    }

    private int getNbTweet()
    {
        try {
            connection =  new URL("https://twitter.com/" + this.fields.get("username")).openConnection();


            BufferedReader in = new BufferedReader(new InputStreamReader(
                    connection.getInputStream(), "UTF-8"));
            String inputLine;
            StringBuilder a = new StringBuilder();
            while ((inputLine = in.readLine()) != null)
                //web.append(inputLine);
                if (inputLine.contains("class=\"ProfileNav-value\""))
                {
                    Matcher matcher = Pattern.compile("\\d+").matcher(inputLine);
                    matcher.find();
                    int i = Integer.valueOf(matcher.group());
                    in.close();
                    return (i);
                }
            in.close();
        }catch ( Exception ex ) {
            ex.printStackTrace();
        }
        return (0);
    }

    public void update() {
        if (this.r == null)
            return ;

        int newNbTweet = getNbTweet();

        if (newNbTweet != this.nbTweet) {

            if (newNbTweet > this.nbTweet)
                this.r.execute();
            this.nbTweet = newNbTweet;
        }
    }
}

File: TEK3/DOTNET/DOT_AREA_2017/Modules/action/twitter/CheckTweet.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Net;
using System.IO;
using System.Text.RegularExpressions;

namespace Modules.action.twitter
{
    class CheckTweet : Action
    {
        WebClient client = new WebClient();
        Dictionary<string, string> fields = new Dictionary<string, string>();
        Reaction r = null;
        string content = null;
        string lastTweetValue = null;
        int n = 0;


        public List<String> getFields()
        {
            return (new List<String>() { { "username" } });
        }

        private int getNbTweets()
        {
            int n = 0;
            try
            {
                content = client.DownloadString("https://twitter.com/" + fields["username"]);
            }
            catch (Exception ex)
            {
                Console.WriteLine(ex);
            }
            using (StringReader reader = new StringReader(content))
            {
                string line;
                while ((line = reader.ReadLine()) != null)
                {
                    if (line.Contains("p class=\"TweetTextSize"))
                    {
                        var s = line.Substring(line.IndexOf(">") + 1);
                        lastTweetValue = "Nouveau tweet : " + s.Substring(0, s.Length - 4);
                        break;
                    }
                }
            }
            using (StringReader reader = new StringReader(content))
            {
                string line;
                while ((line = reader.ReadLine()) != null)
                {
                    if (line.Contains("class=\"ProfileNav-value\""))
                    {
                        n = Int32.Parse(Regex.Match(line, @"\d+").Value);
                        break;
                    }
                }
            }
            return n;
        }

        public void setField(Dictionary<string, string> fields)
        {
            this.fields = fields;
            this.n = getNbTweets();
        }

        public void addReaction(Reaction r)
        {
            this.r = r;
        }

        public void update()
        {
            int newNbTweet = getNbTweets();

            if (newNbTweet != this.n)
            {

                if (newNbTweet > this.n)
                    this.r.execute(lastTweetValue);
                this.n = newNbTweet;
            }
        }
    }
}
